scan/tools/search_tools.py

```diff
@@ -11,37 +9,31 @@ class SearchTools:
     @tool("Search the internet")
     def search_internet(query):
         """Useful to search the internet
-        about a given topic and return relevant results"""
-        return _process_results(query, url="https://google.serper.dev/search")
-
-    @tool("Search news on the internet")
-    def search_news(query):
-        """Useful to search news about a company, stock or any other
-        topic and return relevant results"""
-        return _process_results(query, url="https://google.serper.dev/news")
-
+        about a a given topic and return relevant results"""
+        top_result_to_return = 4
+        url = "https://google.serper.dev/search"
+        payload = json.dumps({"q": query})
+        headers = {"X-API-KEY": os.environ["SERPER_API_KEY"], "content-type": "application/json"}
```

Comment:
`SUPER_API_KEY` could be `None` so check if first.

```suggestion
        if os.environ["SERPER_API_KEY"] is None:
            raise ValueError("The SERPER_API_KEY environment variable must be set")
        
        headers = {"X-API-KEY": os.environ["SERPER_API_KEY"], "content-type": "application/json"}
```